pyproject.toml

```diff
@@ -150,6 +151,46 @@ testing = [
 testing_extra = [
     "torch>=1.7",
 ]
+# needed for gallery examples
+gallery = [
+    "scikit-image[data]",
+    "numpy",
+    "tqdm",
+    "omero-py",
+    "pandas",
+    "zarr",
+    "dask[array,distributed]",
+    "scipy",
+    "magicgui",
+    "tifffile",
+    "matplotlib",
+    "superqt",
+    "imageio",
+    "vispy",
+    "pooch",
+    "nilearn",
+    "xarray",
+]
+# needed to build docs
+docs = [
+    "napari[gallery]",
+    "sphinx<8",
+    "sphinx-autobuild",
+    "sphinx-tabs",
+    "sphinx-tags",
+    "sphinx-design",
+    "sphinx-external-toc",
+    "sphinx-favicon>=1.0",
+    "sphinx-copybutton",
+    "sphinx-gallery",
+    "sphinx_autodoc_typehints==1.12.0",
+    "myst-nb",
+    "napari-sphinx-theme>=0.3.0",
+    "matplotlib",
+    "lxml[html_clean]>5",
+    "imageio-ffmpeg",
+    "pytest",
+]
```

Comment:
With this change, we need to update upgrade dependencies' workflow.

Reply:
@Czaki can you elaborate on this? What do we need to do?

Reply:
Done in ad9fc0161